interfaces/thirdparty/cp2k.py

```diff
@@ -340,64 +343,83 @@ def parse(key, value, indent=''):
         return inp
 
     def _parsemol(self):
-        #make lines shorter
+        # make lines shorter
         inp = self.settings.input.force_eval.subsys
-        #add cell information
+        # add cell information
         nDim = len(self.molecule.lattice)
         keys = ['A', 'B', 'C']
         periodic = ['X', 'XY', 'XYZ']
-        for iDim in range(0,nDim):
-            inp.cell[keys[iDim]] = "{:} {:} {:}".format(*self.molecule.lattice[iDim])
+        for iDim in range(0, nDim):
+            inp.cell[keys[iDim]] = "{:} {:} {:}".format(
+                *self.molecule.lattice[iDim])
         if nDim > 0:
             inp.cell.periodic = periodic[nDim-1]
 
-        #get block of: symbol coords
+        # get block of: symbol coords
         coord_sec = ""
         for atom in self.molecule:
             coord_sec += "\n"
             coord_sec += (" {:}"*4).format(atom.symbol, *atom.coords)
         inp.coord._h = coord_sec
 
     def get_runscript(self):
-        """
-        Run parallel version of Cp2k using srun.
-        """
+        """Run parallel version of Cp2k using srun."""
+        # Try to run cp2k using mpirun and otherwise srun (if available)
+        cp2k_command = self.settings.get("executable", "cp2k.popt")
+
+        # Check the executable name
+        available_executables = (
+            # Serial single core testing and debugging
+            "sdbg",
+            # Serial general single core usage
+            "sopt",
+            # Parallel (only OpenMP), single node, multi core
+            "ssmp",
+            # Parallel (only MPI) multi-node testing and debugging
+            "pdbg",
+            # Parallel (only MPI) general usage, no threads
+            "popt",
+            # parallel (MPI + OpenMP) general usage, threading might improve scalability and memory usage
+            "psmp")
+        if not any((f"cp2k.{x}" == cp2k_command.lower() for x in available_executables)):
```

Comment:
I'd recommend changing ``available_executables`` to a [``set``](https://docs.python.org/3/library/functions.html#func-set) and then only checking the extension of ``cp2k_command``:

```
available_executables: set = {...}

executable = cp2k_command.rsplit('.')[-1]
if executable not in available_executables:
    raise RuntimeError(...)
```

Membership checking (*i.e.* ``set.__contains__()``) is quite a bit faster with sets as compared to builtin Sequence types.

Reply:
It might be better btw to use [``os.path.splitext``](https://docs.python.org/3.8/library/os.path.html#os.path.splitext) instead of [``str.rsplit``](https://docs.python.org/3/library/stdtypes.html#str.rsplit), as the former can handle any path-like object:

```
_, executable = os.path.splitext(cp2k_command)
```

Reply:
thanks!